plugins/techdocs/src/addons/mocks.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export const useTechDocsReaderDom = jest.fn();
+export const useParams = jest.fn();
+
+jest.mock('../reader/components/TechDocsReaderPageContent/dom', () => ({
+  ...(jest.requireActual(
+    '../reader/components/TechDocsReaderPageContent/dom',
+  ) as {}),
+  useTechDocsReaderDom,
```

Comment:
Could you explain why we have to mock `useTechDocsReaderDom` instead of mocking the apis used by it?
If I understand correctly, if we mock the apis, we can move the test-utils to a separate package, along with the plugins framework

Same for `withTechDocsReaderProvider` and `TechDocsStateIndicator`, I have a feeling that we only have to mock apis using TestApiRegistry for this.

Reply:
Before, the dom was passed down to the `TechDocsReaderPage` as a property (see https://github.com/backstage/backstage/pull/10449/files#diff-a437c7c176fd265ecf344553e8668044fb0366ed61ec4da088fabcbd894046edR170). So when another dom was passed through using `withDom`, the component knew about it. The `TechDocsReaderPage` doesn't take in a dom property anymore, therefore we need to mock it. If you have other suggestions, please let me know.

Reply:
I think we are no longer using this version of the Reader page with this dom and the async props, shouldn't we use the same Reader page that we use in the App 🤔 ?  I mean this below, it does not have a dom property: https://github.com/backstage/backstage/blob/techdocs/addon-framework-test-utils/plugins/techdocs/src/reader/components/TechDocsReaderPage/TechDocsReaderPage.tsx

Reply:
yes exactly, thats the one we use. But the TechDocsReaderPageContent component uses the useTechDocsReaderDom hook to get the dom! (which is the content component that has addon support)